bed/compare.go

```diff
@@ -68,6 +116,39 @@ func Compare(a Bed, b Bed) int {
 	return 0
 }
 
+//CompareSize returns zero for beds with an equal length (ChromEnd - ChromStart) and otherwise returns the ordering of the two Bed structs.
+func CompareSize(a Bed, b Bed) int {
+	sizeA := a.ChromEnd - a.ChromStart
+	sizeB := b.ChromEnd - b.ChromStart
+	if sizeA < sizeB {
+		return -1
+	}
+	if sizeA > sizeB {
+		return 1
+	}
+	return 0
+}
+
+//Compare returns zero for beds with an equal ChromEnd position and otherwise returns the ordering of the two Bed entries by chromEnd.
+func CompareChromEnd(a Bed, b Bed) int {
+	if a.ChromEnd < b.ChromEnd {
+		return -1
+	}
+	if a.ChromEnd > b.ChromEnd {
+		return 1
+	}
+	return 0
+}
+
+//CompareChromEndByChrom compares beds by chromosome and then by chromEnd, but not by chromStart.
+func CompareChromEndByChrom(a Bed, b Bed) int {
+	chromComp := strings.Compare(a.Chrom, b.Chrom)
```

Comment:
We should not be using strings.Compare. `a.Chrom == b.Chrom`

Reply:
My main problem is the use of `strings.Compare` instead of the builtin operators. I have pasted the source code for Compare below (comments included). We should stick to builtin operators to achieve the desired logic. 
```
package strings

// Compare returns an integer comparing two strings lexicographically.
// The result will be 0 if a==b, -1 if a < b, and +1 if a > b.
//
// Compare is included only for symmetry with package bytes.
// It is usually clearer and always faster to use the built-in
// string comparison operators ==, <, >, and so on.
func Compare(a, b string) int {
	// NOTE(rsc): This function does NOT call the runtime cmpstring function,
	// because we do not want to provide any performance justification for
	// using strings.Compare. Basically no one should use strings.Compare.
	// As the comment above says, it is here only for symmetry with package bytes.
	// If performance is important, the compiler should be changed to recognize
	// the pattern so that all code doing three-way comparisons, not just code
	// using strings.Compare, can benefit.
	if a == b {
		return 0
	}
	if a < b {
		return -1
	}
	return +1
}
```

Not sure I caught all the times this function is called, but it looks like it is only being used to detect cases where a < b. Only 2 of the 3 comparisons done by the Compare functions are ever used here, so taking the time to evaluate all possible relationships is overkill. You would need to double check the logic is the same, but I think this may be able to be replaced with a simpler function like 
```
func ChromEndLess(a, b bed.Bed) bool {
	if a.Chrom != b.Chrom {
		return a.Chrom < b.Chrom
	}
	return a.ChromEnd < b.ChromEnd
}
```

Reply:
After looking at this and thinking about it, I agree with you.  Would be good to use these basic functions in the future.